homeassistant/config_entries.py

```diff
@@ -817,11 +817,22 @@ def async_get_options_flow(config_entry: ConfigEntry) -> "OptionsFlow":
         raise data_entry_flow.UnknownHandler
 
     @callback
-    def _abort_if_unique_id_configured(self) -> None:
+    def _abort_if_unique_id_configured(
+        self, update_entry: Dict[Any, Any] = None
+    ) -> None:
         """Abort if the unique ID is already configured."""
         if self.unique_id is None:
             return
 
+        if update_entry is not None:
+            for entry in self._async_current_entries():
+                if (
+                    entry.unique_id == self.unique_id
+                    and not update_entry.items() <= entry.data.items()
+                ):
+                    entry.data.update(update_entry)
```

Comment:
We should use the api to update I think, and not update directly.

Reply:
It did consider that, however, it replaces the data dict as a whole. In this case, we want to update it.

Reply:
Ref: <https://github.com/home-assistant/home-assistant/blob/b63dd611b76fb4a1c7fec1a8dbccb2965c917dc0/homeassistant/config_entries.py#L731>

Reply:
However, is should probably add a `_async_schedule_save` call.

Reply:
Copy the data, update it then use the api.

Reply:
Done.